modules/packages/OrderedMap.chpl

```diff
@@ -0,0 +1,746 @@
+    /*
+      Initializes an empty orderedMap containing keys and values of given types.
+
+      :arg keyType: The type of the keys of this orderedMap.
+      :arg valType: The type of the values of this orderedMap.
+      :arg parSafe: If `true`, this orderedMap will use parallel safe operations.
+      :type parSafe: bool
+      :arg comparator: The comparator used to compare elements.
+    */
+    proc init(type keyType, type valType, param parSafe = false,
+              comparator: record = defaultComparator) {
+      _checkKeyType(keyType);
+      _checkValType(valType);
+
+      this.keyType = keyType;
+      this.valType = valType;
+      this.parSafe = parSafe;
+      this.comparator = comparator;
+      this._eltType = (keyType, shared _valueWrapper(valType)?);
+
+      this._set = new orderedSet(_eltType, false, new _keyComparator(comparator)); 
+    }
+
+    /*
+      Initialize this orderedMap with a copy of each of the elements contained in
+      the orderedMap `other`. This orderedMap will inherit the `parSafe` value of 
+      the orderedMap `other`.
+
+      :arg other: An orderedMap to initialize this orderedMap with.
+    */
+    proc init=(const ref other: orderedMap(?kt, ?vt)) lifetime this < other {
+      if !isCopyableType(kt) || !isCopyableType(vt) then
+        compilerError("initializing orderedMap with non-copyable type");
+
+      this.keyType = kt;
+      this.valType = vt;
+      this.parSafe = other.parSafe;
+      this.comparator = other.comparator;
+      this._eltType = (keyType, shared _valueWrapper(valType)?);
+
+      this._set = other._set;
+
+      this.complete();
+    }
+
+    /*
+      Clears the contents of this orderedMap.
+
+      .. warning::
+
+        Clearing the contents of this orderedMap will invalidate all existing
+        references to the elements contained in this orderedMap.
+    */
+    proc clear() {
+      _enter(); defer _leave();
+      _set.clear();
+    }
+
+    /*
+      The current number of keys contained in this orderedMap.
+    */
+    inline proc const size {
+      _enter(); defer _leave();
+      return _set.size;
+    }
+
+    /*
+      Returns `true` if this orderedMap contains zero keys.
+
+      :returns: `true` if this orderedMap is empty.
+      :rtype: `bool`
+    */
+    inline proc const isEmpty(): bool {
+      return size == 0;
+    }
+
+    /*
+      Returns `true` if the given key is a member of this orderedMap, and `false`
+      otherwise.
+
+      :arg k: The key to test for membership.
+      :type k: keyType
+
+      :returns: Whether or not the given key is a member of this orderedMap.
+      :rtype: `bool`
+    */
+    proc const contains(const k: keyType): bool {
+      _enter(); defer _leave();
+      return _set.contains((k, nil));
+    }
+
+    /*
+      Updates this orderedMap with the contents of the other, overwriting the values
+      for already-existing keys.
+
+      :arg m: The other orderedMap
+      :type m: orderedMap(keyType, valType)
```

Comment:
I tried many variants but none of them worked, including the way from Map you mentioned. I just deleted all types since their types are obvious. A wild guess is that it's a bug related to `var comparator: record = defaultComparator`. It's the only difference from Map I see.

Reply:
Possibly you could use `orderedMap(keyType, valType, ?)`?

I'm basing that on `doc/rst/technotes/partialInstantiations.rst`.

If that doesn't work, do you want to open an issue about it?

Reply:
Removing all relevant lines since types are obvious.